torch/csrc/jit/passes/tensorexpr_fuser.cpp

```diff
@@ -625,9 +648,13 @@ class TensorExprFuser {
       REQ(consumer->input(0)->uses().size() == 1);
       REQ(consumer->input(1)->node()->kind() == prim::Constant);
       auto const& listConstruct = consumer->input(0)->node();
-      for (auto const& input : listConstruct->inputs()) {
-        REQ(isFusableOnDevice(input->node()));
-      }
+      // We're merging listconstruct->cat. cat is the consumer and listconstruct
```

Comment:
nit: I wonder if this could be already handled by `canHandle` when we create a TEGroup out of `aten::cat?

Reply:
Yeah, that function could use some TLC, I'll look into it as a follow-up if you don't mind.